packages/lib/server/getLuckyUser.ts

```diff
@@ -302,7 +366,284 @@ export async function getLuckyUser<
+
+      if (queueAndAtributeWeightData?.averageWeightsHosts && queueAndAtributeWeightData?.virtualQueuesData) {
+        attributeWeights = queueAndAtributeWeightData?.averageWeightsHosts;
+        virtualQueuesData = queueAndAtributeWeightData?.virtualQueuesData;
+      }
+    }
+  }
+
+  return { attributeWeights, virtualQueuesData };
+}
+
+async function getQueueAndAttributeWeightData<T extends PartialUser & { priority?: number | null }>(
+  allRRHosts: GetLuckyUserParams<T>["allRRHosts"],
+  routingFormResponse: RoutingFormResponse,
+  attributeWithWeights: AttributeWithWeights
+) {
+  let averageWeightsHosts: { userId: number; weight: number }[] = [];
+
+  const chosenRouteId = routingFormResponse?.chosenRouteId ?? undefined;
+
+  if (!chosenRouteId) return;
+
+  let fieldOptionData: { fieldId: string; selectedOptionIds: string | number | string[] } | undefined;
+
+  const routingForm = routingFormResponse?.form;
+
+  if (routingForm && routingFormResponse) {
+    const response = routingFormResponse.response as FormResponse;
+
+    const routes = zodRoutes.parse(routingForm.routes);
+    const chosenRoute = routes?.find((route) => route.id === routingFormResponse.chosenRouteId);
+
+    if (chosenRoute && "attributesQueryValue" in chosenRoute) {
+      const parsedAttributesQueryValue = children1Schema.parse(chosenRoute.attributesQueryValue);
+
+      const attributesQueryValueWithLabel = (await getAttributesQueryValue())({
+        attributesQueryValue: chosenRoute.attributesQueryValue,
+        attributes: [attributeWithWeights],
+        response,
+        fields: routingFormResponse.form.fields as Fields,
+        getFieldResponse: getFieldResponse,
+      });
+
+      const parsedAttributesQueryValueWithLabel = children1Schema.parse(attributesQueryValueWithLabel);
+
+      if (parsedAttributesQueryValueWithLabel && parsedAttributesQueryValueWithLabel.children1) {
+        averageWeightsHosts = getAverageAttributeWeights(
+          allRRHosts,
+          parsedAttributesQueryValueWithLabel.children1,
+          attributeWithWeights
+        );
+      }
+
+      if (parsedAttributesQueryValue && parsedAttributesQueryValue.children1) {
+        fieldOptionData = getAttributesForVirtualQueues(
+          response,
+          parsedAttributesQueryValue.children1,
+          attributeWithWeights
+        );
+      }
+    }
+  }
+
+  if (fieldOptionData) {
+    return { averageWeightsHosts, virtualQueuesData: { chosenRouteId, fieldOptionData } };
+  }
+
+  return;
+}
+
+function getAverageAttributeWeights<
+  T extends PartialUser & {
+    priority?: number | null;
+    weight?: number | null;
+  }
+>(
+  allRRHosts: GetLuckyUserParams<T>["allRRHosts"],
+  attributesQueryValueChild: Record<
+    string,
+    {
+      type?: string | undefined;
+      properties?:
+        | {
+            field?: any;
+            operator?: any;
+            value?: any;
+            valueSrc?: any;
+          }
```

Comment:
It could be risky, we should use `unknown` instead and safely access these props. It could potentially cause booking by having a run-time error here.

RAQB being a third party library, we don't know the full possibilities of the values here. We could just work on the type of values we understand and ignore those we don't (and log such scenarios maybe)

Reply:
Having a look at this.

Reply:
I think upstream the zod model should be more restrictive, from the code perspective operator/valueSrc isn't needed. it expects value: string[] and field: string. Other values will break the code downstream - most of the validation should happen on the zod side though.